powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehavior.java

```diff
@@ -906,6 +906,14 @@ public PrepareActivationResponse prepareActivation(String activationCode, String
                 throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_INPUT_FORMAT);
             }
 
+            // Ensure presence of the devicePublicKey
+            final String retrievedDevicePublicKey = request.getDevicePublicKey();
+            if (retrievedDevicePublicKey == null || retrievedDevicePublicKey.isEmpty()) {
+                logger.warn("Invalid activation request, activation code: {}", activationCode);
+                // Rollback is not required, error occurs before writing to database
+                throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
```

Comment:
This error code `INVALID_REQUEST` is inconsistent with `INVALID_INPUT_FORMAT` used at https://github.com/wultra/powerauth-server/pull/970/files#diff-a4ce67a9fd8bc9dfacab0302fa5f1cb1a44ee59e9ceafb746ef9e8df877d7492R1162